test/sets.jl

```diff
@@ -6,5 +6,15 @@
             s_copy.weights[1] = 2.0
             @test s.weights[1] == 1.0
         end
+        @testset "IndicatorSet" begin
+            s1 = MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE}(MOI.LessThan(4.0))
+            s2 = MOI.IndicatorSet{MOI.ACTIVATE_ON_ZERO}(MOI.GreaterThan(4.0))
+            s1_copy = copy(s1)
+            s2_copy = copy(s2)
+            @test s1_copy isa MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE}
+            @test s1 == s1_copy
```

Comment:
Check that modifying the set in `s1` does not modify the set in `s1_copy`. You will need a scalar set which is mutable

Reply:
do we even have one? I understand the importance of testing it does not happen, but we are calling copy on the subtest, so if error there is, it's because `copy` is not properly implemented on it

Reply:
@matbesancon `LessThan{BigInt}` is mutable because `BigInt` is mutable. You call also create a dummy mutable coefficient type or a dummy mutable set just for the test.

> we are calling copy on the subtest

That's what we want to test. If you remove this copy on the subtests, the tests still pass while they shouldn't

Reply:
@blegat I played a bit with this idea. Fundamentally, this is breaking an assumption about numbers that will be everywhere in MOI, this PR doesn't change that. `BigInt` being mutable is just a constraint for their construction as far as I know, not supposed to be used.

The implementation from `src/sets`:
```julia
# isbits types, nothing to copy
function Base.copy(set::Union{Reals, Zeros, Nonnegatives, Nonpositives,
                              GreaterThan, LessThan, EqualTo, Interval,
                              SecondOrderCone, RotatedSecondOrderCone,
                              GeometricMeanCone, ExponentialCone,
                              DualExponentialCone, PowerCone, DualPowerCone,
                              PositiveSemidefiniteConeTriangle,
                              PositiveSemidefiniteConeSquare,
                              LogDetConeTriangle, LogDetConeSquare,
                              RootDetConeTriangle, RootDetConeSquare,
                              Integer, ZeroOne, Semicontinuous, Semiinteger})
    return set
end
```

Is then false because it should check `isbits` on the set

Reply:
Good point. We could defer the question of whether we should copy the field of EqualTo... to a separate issue.
Still you might be able to create a mutable abstract set with Float64 to test the fact we ce copy the set for IndicatorSet

Reply:
@blegat see issue #721, and tests added here for `IndicatorSet` copy